dot-net-sdk/store/ConfigurationStore.cs

```diff
@@ -99,8 +99,16 @@ public BanditFlags GetBanditFlags()
         return null;
     }
 
+    private void ClearCaches()
+    {
+        _experimentConfigurationCache.Clear();
+        _banditModelCache.Clear();
+        _banditFlagCache.Clear();
+    }
     public void FetchConfiguration()
     {
+        ClearCaches();
+
         FlagConfigurationResponse experimentConfigurationResponse = Get();
         experimentConfigurationResponse.Flags.ToList()
             .ForEach(x => { this.SetExperimentConfiguration(x.Key, x.Value); });
```

Comment:
Between clearing the cache and populating it again, the `getAssignments` methods will not have configurations to read from. This is obvious a very short time period but will happen. In some languages setting a variable is idempotent, is it that way in dotnet too?

```
const tempFlagsArray = Get()

ExperimentConfigurations = tempFlagsArray
```

Reply:
Added a lock to ensure caches cannot be read while they're being reset.